File: pset6/readability/readability.py

text = input("Text: ")

words = 1
sentences = 0
letters = 0

for char in text :
    if(char == "!" or char == "?" or char ==".") :
        sentences += 1
    elif(char == " ") :
        words += 1
    elif(char.isalpha() == True) :
        letters += 1

L = 100.0 * (float(letters) / float(words))
S= 100.0 * (float(sentences) / float(words) )

grade = 0.0588 * L - 0.296 * S - 15.8
grade = round(grade)

if(grade >= 16 ) :
    print("Grade 16+")
elif(grade < 1) :
    print("Before Grade 1")
else :
    print(f"Grade {grade}")
    

File: pset2/readability/readability.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
#include <cs50.h>
int main() {
    string text = get_string("Text:\n");
    int  words = 1;
    int  sentences = 0;
    int  letters = 0;
    for (int i = 0, n = strlen(text); i <= n; i++) {
        if (isalpha(text[i])) {
            letters += 1;
        }
        else if (text[i] == '!' || text[i] == '?' || text[i] == '.') {
            sentences += 1;
        }
        else if (text[i] == ' ') {
            words += 1;
        }
    }

    float  L =  100 * ((float)(letters) / (float)(words));
    float S = 100 * ((float)(sentences) / (float)(words));
    double index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);
    if (index > 16) {
        printf("Grade 16+\n");
    }
    else if (index < 1) {
        printf("Before Grade 1\n");
    }
    else {
        printf("Grade %i\n", (int)(index));
    }
    return 0;
}